WordPress/gutenberg
# Make the custom CSS validation error message accessible

Fixes https://github.com/WordPress/gutenberg/issues/56437

This PR changes the custom CSS validation error message to make it accessible. On top of that, I'd like to add a few considerations:
- Using ad-hoc components that are only used once in a single place defeats the entire purpose of building reusable components.
- Instead, using the existing components makes sure that there is design consistency and that the components are accessible, as they are supposed to be designed, built, and tested for accessibility.
- I'd recommend to not introduce ad-hoc, single-use, components in the future. If there is a proven need for a new UI components that covers a vlaid use case and can be reused across the editor, I'd suggest to design a new component to be included in the `components` package. Only by doing that the new component would be guaranteed to be consistent, reusable, tested, and accessible.



## What?

The custom CSS validation error message is not accessible. It is also a unique UI in the editor, there's no other usage of a red icon with tooltip and I'd argue it's am unexpected pattern for users.

## Why?

Any user-input validation message must be accessible.

## How?

Replaces the current implementation based on a red icon plus tooltip with the standard Notice component.
- The Notice is set to error instead of warning, because this is an error.
- The Notice will produce by. default an accessible audible message with `assertive` politeness level.
- I'm placing the Notice at the top of the textarea because it's more visible in that spot.

## Testing Instructions




- Follow the reproduction instructions from the issue https://github.com/WordPress/gutenberg/issues/56437
- Observe that instead of a red icon at the bottom right of the textarea, an error Notice is shown before the textarea.

### Testing Instructions for Keyboard


## Screenshots or screencast 

Before and after:

![before and after](https://github.com/WordPress/gutenberg/assets/1682452/2f7ca6d1-a54f-4e59-ae2b-535e88e52a43)
